scripts/globals/mobskills/self-destruct_321.lua

```diff
@@ -13,9 +13,13 @@ mobskill_object.onMobSkillCheck = function(target, mob, skill)
 end
 
 mobskill_object.onMobWeaponSkill = function(target, mob, skill)
-    local amount = 9999 * skill:getTotalTargets()
-    local dmg = xi.mobskills.mobFinalAdjustments(amount, mob, skill, target, xi.attackType.PHYSICAL, xi.damageType.FIRE, xi.mobskills.shadowBehavior.WIPE_SHADOWS)
-    target:takeDamage(dmg, mob, xi.attackType.PHYSICAL, xi.damageType.FIRE)
+    local amount = 9999
+
+    local info = xi.mobskills.mobMagicalMove(mob, target, skill, amount, xi.magic.ele.FIRE, 2.1, xi.mobskills.magicalTpBonus.MAB_BONUS, 1)
+    local dmg = xi.mobskills.mobFinalAdjustments(info.dmg, mob, skill, target, xi.attackType.MAGICAL, xi.damageType.FIRE, xi.mobskills.shadowBehavior.IGNORE_SHADOWS)
+
+    mob:setHP(1)
```

Comment:
Curious why setting HP to 1 here? Wouldn't that leave it at 1 HP if there's no DOTs and everyone in the fight is dead from this going off?

Reply:
As seen in captures, the self destruct doesn't kill the bomb. It does however reduce its HP to 0. I think this is used to prevent the battlefield from being won.

Reply:
https://www.youtube.com/watch?v=SLYUyRnUmpA&ab_channel=WiggoCaptures
5:46 you can see this happening

Reply:
I do see that its killing the bomb, it just doesn't die until after the animation for the self-destruct completes.

Reply:
yes yes. without setting HP to 1 it kills him immediately. Could I just use the unkillable tag to achieve this. In time bomb's lua I'm actually handling him to be unkillable, using selfdestruct, then killing him. Should I just handle it all in the lua?

Reply:
This should be fine then. Just wanted to make sure. :D